backend/coins/eth/account.go

```diff
@@ -379,23 +379,44 @@ func (account *Account) update() error {
 		}
 	}
 
+	var balance *big.Int
 	if account.coin.erc20Token != nil {
-		balance, err := account.coin.client.ERC20Balance(account.address.Address, account.coin.erc20Token)
-		if err != nil {
-			return errp.WithStack(err)
-		}
-		account.balance = coin.NewAmount(balance)
+		balance, err = account.coin.client.ERC20Balance(account.address.Address, account.coin.erc20Token)
 	} else {
-		balance, err := account.coin.client.Balance(context.TODO(), account.address.Address)
-		if err != nil {
-			return errp.WithStack(err)
-		}
-		account.balance = coin.NewAmount(balance)
+		balance, err = account.coin.client.Balance(context.TODO(), account.address.Address)
+	}
+
+	if err != nil {
+		return errp.WithStack(err)
 	}
 
+	pendingAmount := pendingTxsAmount(outgoingTransactionsData)
+	account.balance = coin.NewAmount(balance.Sub(balance, pendingAmount))
+
 	return nil
 }
 
+// pendingTxsAmount returns the total amount of pending transactions, including fees (only for ETH txs).
+func pendingTxsAmount(outgoingTransactionsData []*accounts.TransactionData) *big.Int {
+	pendingTxAmount := big.NewInt(0)
+	for _, tx := range outgoingTransactionsData {
+		if tx.Status == "pending" {
+			// Skip sendSelf txs
+			if tx.Type == "send" {
+				pendingTxAmount = pendingTxAmount.Add(pendingTxAmount, tx.Amount.BigInt())
+			}
+			if !tx.FeeIsDifferentUnit {
+				// tx Fee is considered only for ETH transactions. For ERC20 tokens it should
+				// be detracted to the balance of the related ETH account. This is not done at
```

Comment:
I think you mean subtracted not detracted :smile:

Reply:
TIL: detract is an English "false friend". Didn't know that :rofl: